src/lib/Form/DataTransformer/DateTimeFormatTransformer.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Form\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+use Symfony\Component\Form\Exception\TransformationFailedException;
+use EzSystems\EzPlatformAdminUi\Service\UserPreferences\DateTimeFormat\DateTimeFormatTransformer as Transformer;
+
+class DateTimeFormatTransformer implements DataTransformerInterface
+{
+    /**
+     * @var \EzSystems\EzPlatformAdminUi\Form\DataTransformer\DateTimeFormatTransformer
+     */
+    private $transformer;
+
+    /**
+     * @param string[] $allowedDateFormats
+     * @param string[] $allowedTimeFormats
+     */
+    public function __construct(array $allowedDateFormats = [], array $allowedTimeFormats = [])
+    {
+        $this->transformer = new Transformer($allowedDateFormats, $allowedTimeFormats);
```

Comment:
You are using `$allowedDateFormats` and `$allowedTimeFormats` only to initialize `Transformer` object. This is a candidate for dependency injection.